### PR TITLE
Added Tests

### PR DESCRIPTION
**Pull request by sshestaka**
1. Added test cases for Book and Category entities (repository, service, controller).

Added test cases for Book and Category entities

### CODE REVIEW

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper = new ObjectMapper();
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]: How it possible??? What about DI?

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]: Why protected?

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        requestDto.setPrice(BigDecimal.valueOf(44.95));
+        requestDto.setDescription("Test-Description");
+        requestDto.setCoverImage("Test-Cover-Image");
+        BookDto expected = new BookDto()
+                .setTitle(requestDto.getTitle())
+                .setAuthor(requestDto.getAuthor())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice())
+                .setDescription(requestDto.getDescription())
+                .setCoverImage(requestDto.getCoverImage());
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java:138]: Save it in templates and you can reuse it anywhere

File: src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        );
+        expected.add(
+                new BookDto()
+                        .setId(3L)
+                        .setTitle("Black Book")
+                        .setAuthor("Black Author")
+                        .setIsbn("Black-ISBN")
+                        .setPrice(BigDecimal.valueOf(21.99))
+                        .setDescription("Black description")
+                        .setCoverImage("Black cover image")
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/BookControllerTest.java]: Write templates

File: src/test/java/mate/academy/onlinebookstore/controller/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    @Test
+    @DisplayName("Get list of all categories")
+    @WithMockUser(username = "user", roles = {"USER"})
+    void getAll_GivenCategoriesInCatalog_ShouldReturnAllCategories() throws Exception {
+        List<CategoryDto> expected = new ArrayList<>();
+        expected.add(new CategoryDto().setId(1L).setName("Category1").setDescription("Category1"));
+        expected.add(new CategoryDto().setId(2L).setName("Category2").setDescription("Category2"));
+        expected.add(new CategoryDto().setId(3L).setName("Category3").setDescription("Category3"));
```
Comment (Sanderfuck) [inline src/test/java/mate/academy/onlinebookstore/controller/CategoryControllerTest.java]: Use constants for strings

**Review by Sanderfuck — CHANGES_REQUESTED** [review]
Check coverage for methods and lines

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**